### PR TITLE
Do not assign the author of a PR as the reviewer

### LINKED ISSUE #96: Prevent highfive from assigning PRs to PR author
Right now, if a collaborator opens a new PR, there is a chance that he/she will be randomly assigned to review it. We should check to make sure that the PR author isn't being selected as the reviewer for his/her own PR.

### PR DESCRIPTION
**Pull request by aneeshusa**
Fixes #96. I want to add myself as a collaborator for saltfs, but I also author most of the PRs there...

### CODE REVIEW

**Comment by jdm** [comment]
FYI, highfive doesn't run on saltfs right now.

File: handlers/assign_reviewer/__init__.py [hunk 1]
```diff
@@ -18,6 +18,21 @@ def find_reviewer(comment, collaborators):
     return None
 
 
+def choose_reviewer(pr, collaborators):
+    """
+    Choose a (psuedo-)random reviewer from the collaborators, who is not the
```
Comment (jdm) [inline handlers/assign_reviewer/__init__.py]: `pseudo`

File: handlers/assign_reviewer/__init__.py [hunk 2]
```diff
@@ -18,6 +18,21 @@ def find_reviewer(comment, collaborators):
...
+    author. If there are no collaborators other than the author, return None.
+    """
+    author = pr['user']['login']
+    # Avoid using sets to maintain the order of reviewers and
+    # thus ensure an even rotation of reviewers
+    potential_reviewers = [collaborator for collaborator in collaborators
+                           if collaborator != author]
+    if len(potential_reviewers) == 0:
```
Comment (jdm) [inline handlers/assign_reviewer/__init__.py]: `if not potential_reviewers:`

File: handlers/assign_reviewer/tests/new_pr_author_is_only_collaborator.json [hunk 3]
```diff
@@ -0,0 +1,430 @@
+{
+  "initial": {
+  },
+  "expected": {
+  },
```
Comment (jdm) [inline handlers/assign_reviewer/tests/new_pr_author_is_only_collaborator.json]: This should have `assignee: null`, or it won't check anything.

**Comment by aneeshusa** [comment]
What's the process to add saltfs to highfive? I've been filing E-Easy issues to try and attract newcomers, and last week we got two new contributors, so I'd love to see highfive added.

**Comment by jdm** [comment]
It should be a matter of adding a webhook pointing at `http://www.joshmatthews.net/highfive/newpr.py` that sees issues and pull requests.

File: handlers/assign_reviewer/tests/new_pr_author_is_only_collaborator.json (hunk 3 above)
Comment (aneeshusa) [inline handlers/assign_reviewer/tests/new_pr_author_is_only_collaborator.json]: The test fails with `None == null` when I add this, and I'm not sure why.

Comment (aneeshusa) [inline handlers/assign_reviewer/tests/new_pr_author_is_only_collaborator.json]: Figured it out - I was setting `"assignee": "null"`.

**Comment by jdm** [comment]
Thanks!

**Pull request merged**